### DISCUSSION TITLE
Email is not populated after user created using email and password sign in and sign up

### DISCUSSION THREAD

**Discussion by myhendry** [supabase/supabase#29362] [Questions] [answered]
Hi, i use email and password login. the user can successfully sign up and sign in, however, the trigger doesn't populate my email field in my user table. it only populate my id field. anyone knows how i can solve it? below is my supabase function. i looked at it several times but it seems fine

```
begin
  insert into public.users (id, email, full_name, avatar_url)
  values (new.id, new.raw_user_meta_data->>'email', new.raw_user_meta_data->>'full_name', new.raw_user_meta_data->>'avatar_url');
  return new;
end;

```

my supabase has a user table with the email field

**Comment by GaryAustin1** [comment, marked as answer]
You should use new.email

**Reply by myhendry** [reply]
Thanks! that works. i was referring to a youtube tutorial and he was using new.raw_user_meta_data->>'email' with google and github auth. may i know if there was a change in the api? or new.raw_user_meta_data->>'email' only works with google and github auth but not email with password auth? i ran select raw_user_meta_data from auth.users; and it does return me an object with the email record.

**Reply by GaryAustin1** [reply]
New.email has always been the way. The user metadata way was added later. I've not studied what providers it works for. Also several things get done during updates to the table after the insert, or at least they did 6 months or so ago when I last looked at it.

**Reply by myhendry** [reply]
i see, thanks Gary! i ll be adding in Github and Google Oauth to my app later. I will see if any difference to email with password auth and will post an update here later. Once again, thanks a lot for helping
